src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -956,78 +1018,89 @@ async_manifest_view::time_based_retention(
         auto res = co_await get_cursor(
           _stm_manifest.get_archive_start_offset(),
           model::prev_offset(_stm_manifest.get_start_offset().value()));
-        if (res.has_failure() && res.error() != error_outcome::out_of_range) {
-            vlog(
-              _ctxlog.error,
-              "Failed to compute time-based retention {}",
-              res.error());
-            co_return res.as_failure();
+        if (res.has_failure()) {
+            if (res.error() == error_outcome::out_of_range) {
+                // The cutoff point is outside of the offset range, no need to
+                // do anything
+                vlog(
+                  _ctxlog.debug,
+                  "There is no segment old enough to be removed by retention");
+                co_return result;
+            } else {
```

Comment:
style-nit: can't reduce indention by removing the else after an early return. Same below

Reply:
Yeah, I prefer this style when dealing with important code paths (like retention). Don't really mind either way though.